patches/server/1060-Brigadier-based-command-API.patch

```diff
@@ -0,0 +1,2506 @@
++        return this.of(RangeArgument.floatRange(), type -> VanillaArgumentProviderImpl.convertToRange(type, doubleRange -> () -> doubleRange));
++    }
++
++    private static <C extends Number & Comparable<C>, T extends RangeProvider<C>> T convertToRange(MinMaxBounds<C> bounds, Function<Range<C>, T> converter) {
++        if (bounds.isAny()) {
++            return converter.apply(Range.all());
++        } else if (bounds.min().isPresent() && bounds.max().isPresent()) {
++            return converter.apply(Range.closed(bounds.min().get(), bounds.max().get()));
++        } else if (bounds.max().isPresent()) {
++            return converter.apply(Range.atMost(bounds.max().get()));
++        } else if (bounds.min().isPresent()) {
++            return converter.apply(Range.atLeast(bounds.min().get()));
++        }
++        throw new IllegalStateException("This is a bug: " + bounds);
++    }
++
++    @Override
++    public ArgumentType<World> world() {
++        return this.of(DimensionArgument.dimension(), dimensionLocation -> {
++            // based on DimensionArgument#getDimension
++            ResourceKey<Level> resourceKey = ResourceKey.create(Registries.DIMENSION, dimensionLocation);
++            @Nullable ServerLevel serverLevel = MinecraftServer.getServer().getLevel(resourceKey);
++            if (serverLevel == null) {
++                throw DimensionArgument.ERROR_INVALID_VALUE.create(dimensionLocation);
++            } else {
++                return serverLevel.getWorld();
++            }
++        });
++    }
++
++    @Override
++    public ArgumentType<GameMode> gameMode() {
++        return this.of(GameModeArgument.gameMode(), type -> requireNonNull(GameMode.getByValue(type.getId())));
++    }
++
++    @Override
++    public ArgumentType<HeightMap> heightMap() {
++        return this.of(HeightmapTypeArgument.heightmap(), CraftHeightMap::fromNMS);
++    }
++
++    @Override
++    public ArgumentType<UUID> uuid() {
++        return this.of(UuidArgument.uuid(), (identity) -> identity);
++    }
++
++    @Override
++    public ArgumentType<Criteria> objectiveCriteria() {
++        return this.of(ObjectiveCriteriaArgument.criteria(), CraftCriteria::getFromNMS);
++    }
++
++    @Override
++    public ArgumentType<LookAnchor> entityAnchor() {
++        return this.of(EntityAnchorArgument.anchor(), type -> LookAnchor.valueOf(type.name()));
++    }
++
++    @Override
++    public ArgumentType<Integer> time(final int minTicks) {
++        return this.of(TimeArgument.time(minTicks), identity -> identity);
++    }
++
++    @Override
++    public ArgumentType<Mirror> templateMirror() {
++        return this.of(TemplateMirrorArgument.templateMirror(), mirror -> Mirror.valueOf(mirror.name()));
++    }
++
++    @Override
++    public ArgumentType<Rotation> templateRotation() {
++        return this.of(TemplateRotationArgument.templateRotation(), mirror -> Rotation.valueOf(mirror.name()));
```

Comment:
This argument type is broken, the api isn't a 1:1 representation either i'm not sure if it's worth to include this one for now.

Reply:
Thanks